src/spi.rs

```diff
@@ -781,6 +824,69 @@ where
     }
 }
 
+pub struct DmaBuilder<SPI> {
+    spi: SPI,
+}
+
+pub struct Tx<SPI> {
+    spi: PhantomData<SPI>,
+}
+
+pub struct Rx<SPI> {
+    spi: PhantomData<SPI>,
+}
+
+impl<SPI> DmaBuilder<SPI>
+where
+    SPI: Deref<Target = spi1::RegisterBlock>,
+{
+    pub fn tx(self) -> Tx<SPI> {
+        self.new_tx()
+    }
+
+    pub fn rx(self) -> Rx<SPI> {
+        self.new_rx()
+    }
+
+    pub fn txrx(self) -> (Tx<SPI>, Rx<SPI>) {
+        (self.new_tx(), self.new_rx())
+    }
+
+    fn new_tx(&self) -> Tx<SPI> {
+        self.spi.cr2.write(|w| w.txdmaen().enabled());
+        Tx { spi: PhantomData }
+    }
+
+    fn new_rx(self) -> Rx<SPI> {
+        self.spi.cr2.write(|w| w.rxdmaen().enabled());
```

Comment:
```suggestion
        self.spi.cr2.modify(|_, w| w.rxdmaen().enabled());
```
... or else txrx() doesn't work because txdmaen is reset